### PR TITLE
Form.Item support layout

### PR DESCRIPTION
**Pull request by crazyair**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
https://github.com/ant-design/ant-design/discussions/47500
https://github.com/ant-design/ant-design/issues/48951
https://github.com/ant-design/ant-design/issues/48951

### 📝 Changelog

由于 inline 需要父级控制，固 `Form.Item` 只支持水平、垂直

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Form.Item support layout     |
| 🇨🇳 Chinese |    Form.Item 支持 layout       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/form/FormItem/ItemHolder.tsx [hunk 1]
```diff
@@ -99,6 +101,9 @@ export default function ItemHolder(props: ItemHolderProps) {
     [`${itemPrefixCls}-has-error`]: mergedValidateStatus === 'error',
     [`${itemPrefixCls}-is-validating`]: mergedValidateStatus === 'validating',
     [`${itemPrefixCls}-hidden`]: hidden,
+
+    // Layout
+    [`${prefixCls}-item-${layout}`]: layout,
```
Comment (MadCcc) [inline components/form/FormItem/ItemHolder.tsx:106]:
```suggestion
    [`${itemPrefixCls}-${layout}`]: layout,
```

Comment (crazyair) [reply components/form/FormItem/ItemHolder.tsx]: done

File: components/form/FormItemLabel.tsx [hunk 2]
```diff
@@ -56,11 +57,11 @@ const FormItemLabel: React.FC<FormItemLabelProps & { required?: boolean; prefixC
   required,
   requiredMark,
   tooltip,
+  vertical,
 }) => {
   const [formLocale] = useLocale('Form');
 
   const {
-    vertical,
```
[the remark this answers is not in the document]
Comment (crazyair) [reply components/form/FormItemLabel.tsx:63]: 在调用该组件的地方合并了

**Review by MadCcc — APPROVED** [review]
LGTM

**Comment by zhenzhenChange** [comment]
文档标错了吧

image

**Comment by crazyair** [comment]
> 文档标错了吧
> 
> image

Thanks♪(･ω･)ﾉ

**Comment by yh20421** [comment]
> > 文档标错
> > 图像
> 
> 谢谢♪(･ω･)ﾉ

应该删第一个layout  第二个带版本号限制的删掉了

**Comment by crazyair** [comment]
> > > 文档标错
> > > 图像
> > 
> > 
> > 谢谢♪(･ω･)ﾉ
> 
> 应该删第一个layout 第二个带版本号限制的删掉了

我看了没问题，Form layout 是早就有的，Form.Item layout 是 5.1.8.0 加的

**Pull request merged**